src/application/services/useNote.ts

```diff
@@ -166,15 +167,24 @@ export default function (options: UseNoteComposableOptions): UseNoteComposableSt
    * @param parentId - Id of the parent note. If null, then it's a root note
    */
   async function save(content: NoteContent, parentId: NoteId | undefined): Promise<void> {
+    const { userEditorTools } = useAppState();
+
     if (note.value === null) {
       throw new Error('Note is not loaded yet');
     }
 
+    /**
+     * Format specified tools
+     */
+    const specifiedNoteTools = content.blocks.map((block) => {
+      return { name: block.type, id: (userEditorTools.value ?? []).find((tool) => tool.name === block.type)!.id };
```

Comment:
note can contain tools that are not installed by the user. For example, there are 2 collaborators on the note. One has list plugin installed, the other does not. If the second user tries to update the note, what will happen?